core/sparse_array.cpp

```diff
@@ -46,6 +47,25 @@ SparseArray::SparseArray(unsigned int p_nrows, unsigned int p_ncols)
 	
 	row_offsets_[nrows_set_] = 0;
 	finished_ = false;
+
+    //Allocate space for 2-D vectors
+    //Columns and Distances are now 2-D vectors with each row corresponding to one focal individual. This negates the need for row offsets.
+
+    initial_width=5000;       //Some random initial width for each row. Can be changed later.
```

Comment:
I'd say start with 100, then double it when you run out.

Reply:
Yes; the width of a row is the number of other individuals that one focal individual interacts with, and that will often be low even for large model, if the interaction distance is short.  Right now all of these buffers are re-allocated in every generation and grow up to their target size from scratch, over and over as the model runs.  Ideally the grown buffers, at the size that turns out to be sufficient for the model's dynamics, would be kept and re-used over time, so that once the code equilibrates no buffers ever need to be allocated or grown again.  Not sure whether that additional complication would be worth the work, though; we'll have to see whether malloc/realloc show up as significant overhead with this redesign.  For now 100 seems reasonable – but for some models it might already be too high, and might substantially increase the overhead for the spare array (which can be the bulk of all memory usage for large spatial models, I think).